File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int del = 0;
        int b = 0;
        for (char c : s){
            if (c == 'b'){
                ++b;
            } else {
                del = min(del + 1, b);
            }
        }
        return del;
    }
};

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        n = len(s)
        dp = [0] * (n + 1)
        b = 0
        for i, char in enumerate(s, 1):
            if char == 'b':
                dp[i] = dp[i - 1]
                b += 1
            else:
                dp[i] = min(dp[i - 1] + 1, b)
        return dp[n]